presto-orc/src/main/java/com/facebook/presto/orc/stream/BooleanInputStream.java

```diff
@@ -125,6 +125,30 @@ public int countBitsSet(int items)
         return count;
     }
 
+    /**
+     * Sets the vector element to true if the bit is set.
+     */
+    public void getSetBits(int batchSize, boolean[] vector)
+            throws IOException
+    {
+        for (int i = 0; i < batchSize; i++) {
+            vector[i] = nextBit();
+        }
+    }
+
+    /**
+     * Sets the vector element to true if the bit is set, skipping the null values.
+     */
+    public void getSetBits(int batchSize, boolean[] vector, boolean[] isNull)
```

Comment:
I looked through that code, and it has not yet been applied to MapStreamReader.  I think that is because the code still needs to get the lengths first to figure out how many keys/values to read from the underlying streams.  nextIntVector still has an implementation that takes isNull because of this.  Since the inMap stream replaces the lengths stream I think I still need to have this.